vendor/engines/split_accounts/app/models/split_accounts/split_account.rb

```diff
@@ -27,5 +29,16 @@ def extra_penny_count
       splits.select(&:extra_penny?).size
     end
 
+    # Stopped using SQL because that didn't seem to work until the built splits
+    # and subaccounts were created.
+    #
+    # SQL version:
+    #   subaccounts.where("expires_at IS NOT NULL").order(expires_at: :asc).first
+    #
+    def earliest_expiring_subaccount
+      subaccounts = splits.map{ |split| split.subaccount }.select(&:expires_at)
```

Comment:
I would expect `subaccounts` to work even with unpersisted splits, but I guess it might have to do with how the splits are built. If they're built as `Split.new(parent: parent, subaccount: subaccount)`, I could see how it wouldn't work, but if you do `parent.splits.build(subaccount: subaccount)`, I would expect it to. I'm guessing that it's the nested form gem that's taking care of the building. Maybe if it's doing `parent.splits.build(subaccount_id: id)`, that's not working.

Not using the `has_many :through` is fine, it's just that this is a little hard to read. 

`select(&:expires_at)` confused me a little bit thinking it was doing a kind of map and returning a bunch of dates. All accounts should have an expiration date, so is that check even necessary?

`splits.map(&:subaccount).min_by(&:expires_at)`?